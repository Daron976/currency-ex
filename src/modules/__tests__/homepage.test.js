/*eslint-disable*/
import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Homepage from '../homepage';

describe('Homepage', () => {
  test('Homepage items renders as expected', async () => {
    const snapHomepage = renderer.create(
      <BrowserRouter>
        <Homepage />
      </BrowserRouter>,
    )
      .toJSON();
    expect(snapHomepage).toMatchSnapshot();
  });
  test('Proceed is present', async () => {
    render(
      <BrowserRouter>
        <Homepage />
      </BrowserRouter>,
    );

    const btn = screen.getAllByText('Proceed');

    expect(btn).toBeInTheDocument;
  });
  test('Proceed button works', async () => {
    render(
      <BrowserRouter>
        <Homepage />
      </BrowserRouter>,
    );

    const btn = screen.getByText('Proceed');

    fireEvent.click(btn);

    expect(btn).not.toBeInTheDocument;
  });
});
