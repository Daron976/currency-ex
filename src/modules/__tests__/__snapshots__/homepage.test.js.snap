// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Homepage items renders as expected 1`] = `
<section
  className="home-page"
>
  <img
    alt="App logo"
    src="https://img.icons8.com/external-icons-smashing-stocks/68/000000/external-Currencies-stock-market-icons-smashing-stocks.png"
  />
  <div>
    <h1>
      Welcome to Currex
    </h1>
  </div>
  <a
    href="/continents"
    onClick={[Function]}
  >
    <button
      id="proceed"
      name="proceed"
      type="button"
    >
      Proceed
    </button>
  </a>
</section>
`;
